include/boost/format/format_class.hpp

```diff
@@ -161,8 +161,8 @@ namespace boost {
         void make_or_reuse_data(std::size_t nbitems);// used for (re-)initialisation
 
         // member data --------------------------------------------//
-        std::vector<format_item_t>  items_; // each '%..' directive leads to a format_item
-        std::vector<bool> bound_; // stores which arguments were bound. size() == 0 || num_args
+        std::vector<format_item_t,typename Alloc::template rebind<format_item_t>::other >  items_; // each '%..' directive leads to a format_item
+        std::vector<bool, typename Alloc::template rebind<bool>::other > bound_; // stores which arguments were bound. size() == 0 || num_args
```

Comment:
This isn't right, because you can't assume that an Allocator has a member template `rebind`. In fact with `std::allocator` this will not even compile in C++20 where the `rebind` member was removed. Instead of `typename A::template rebind<T>::other` use `typename boost::allocator_rebind<A, T>::type`. Include `<boost/core/allocator_access.hpp>`.